mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulVisitedProfileScreen.kt

```diff
@@ -5,38 +5,64 @@ import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
 import ch.epfl.sdp.mobile.application.Profile
 import ch.epfl.sdp.mobile.application.PuzzleId
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
 import ch.epfl.sdp.mobile.application.chess.ChessFacade
 import ch.epfl.sdp.mobile.ui.profile.ProfileScreen
 import ch.epfl.sdp.mobile.ui.profile.ProfileScreenState
 import ch.epfl.sdp.mobile.ui.profile.VisitedProfileScreenState
 import ch.epfl.sdp.mobile.ui.social.ChessMatch
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
 
 /**
  * An implementation of the [VisitedProfileScreenState] that performs a given profile's [ChessMatch]
  * requests.
  *
+ * @param currentUser current logged in user
  * @param user the given [Profile].
  * @param actions the [ProfileActions] which are available on the screen.
- * @param onGameItemClickAction callback if challenge button clicked
+ * @param onChallengeClickAction callback if challenge button clicked
  * @param chessFacade the [ChessFacade] used to perform some requests.
  * @param scope the [CoroutineScope] on which requests are performed.
  */
 class FetchedUserProfileScreenState(
+    private val currentUser: AuthenticatedUser,
     private val user: Profile,
     actions: State<ProfileActions>,
-    onGameItemClickAction: State<(String) -> Unit>,
```

Comment:
Question: Why remove this callback? Is it not needed for clicking on games in the player's history?

Reply:
This was still there because we used it as a dummy for the `Unfolow` Button. The logic of the onGameItemClick is in the actions arguement

Reply:
I see!